### PR TITLE
Fix possible IndexError during shutdown

### LINKED ISSUE #5390: scheduler: traceback on shutdown
```
    Traceback (most recent call last):                                                                
      File "/cylc/flow/scheduler.py", line 1751, in shutdown
        await self._shutdown(reason)
      File "/cylc/flow/scheduler.py", line 1836, in _shutdown    
        self.run_event_handlers(self.EVENT_SHUTDOWN, reason.args[0])
    IndexError: tuple index out of range
```

The reason being that `reason.args` has length `0`. This logic should be made more robust.

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #5390

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes
- [x] No tests needed - cba
- [x] No changelog entry needed (very minor)
- [x] No docs needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/exceptions.py [hunk 1]
```diff
@@ -430,7 +430,6 @@ class NoHostsError(CylcError):
     """None of the hosts of a given platform were reachable."""
     def __init__(self, platform):
         self.platform_name = platform['name']
-        super().__init__()
```
Comment (oliver-sanders) [inline cylc/flow/exceptions.py:433]: Ok, funky.

It looks like `HostSelectException` might be similar?

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/exceptions.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/exceptions.py:433]: Good spot, I have fixed that too now

**Review by oliver-sanders — APPROVED** [review]

**Comment by hjoliver** [comment]
Maybe I'm being dense but how do the code changes on this branch fix the IndexError during shutdown?

**Comment by MetRonnie** [comment]
The traceback:
```
    self.run_event_handlers(self.EVENT_SHUTDOWN, reason.args[0])
IndexError: tuple index out of range
```
The code:
```py
class SomethingError(CylcError):
    def __init__(self, platform):
        self.platform_name = platform['name']
        super().__init__()
```
Calling `super().__init__()` means the exception has an empty list for the `args` property

**Review by hjoliver — APPROVED** [review]

**Pull request merged**